### PR TITLE
Incentivized Liquidation System for Leverage and Perpetual Market

### PR DESCRIPTION
**Pull request by amityadav0**
## Description

- Add spec for discussion

### What has Changed?

What specific problem were you aiming to address, and how did you successfully resolve it? If tests were not uploaded for this pull request or if coverage decreased, please provide an explanation for the change.

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/elys-network/elys/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/elys-network/elys/blob/main/docs/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/elys-network/elys/blob/main/CONTRIBUTING.md#testing)
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items._

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### Deployment Notes

Are there any specific considerations to take into account when deploying these changes? This may include new dependencies, scripts that need to be executed, or any aspects that can only be evaluated in a deployed environment.

### Screenshots and Videos

Please provide any relevant before and after screenshots by uploading them here. Additionally, demo videos can be highly beneficial in demonstrating the process.

### CODE REVIEW

File: x/leveragelp/spec/incentivized_liquidation_system.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
+
+# Incentivized Liquidation System for Leverage Market
+
+## Diagram
+
+```plaintext
++------------+                  +------------+                  +-------------+                 
+|            |                  |            |                  |             |                
+| Depositors |                  | Borrowers  |                  | Liquidators |
```
Comment (cosmic-vagabond) [inline x/leveragelp/spec/incentivized_liquidation_system.md:9]: @amityadav0 as we are managing both liquidation and SL flows, what do you think of using a different name for "Liquidators" such as Watchers or Sentinels?

File: x/leveragelp/spec/incentivized_liquidation_system.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
++-----v------+                  +-----v------+                  +-----v-------+                   
+|            |                  |            |                  |             |                 
+|  Stable    |<-----------------| Leverage   |<-----------------| Liquidation |
+|  Coin Pool |   Borrow coins   | Position   |   Monitor        |   Function  |   Monitor        
+|  Module    |                  | Module     |   positions      |             |   positions       
++------------+                  +------------+                  +-------------+                  
+      |                               |                               |                               
+      |                               |                               |                                
+      |  Interest                      | Position health calculated   | If health < threshold, or certain price is met(stop loss)     
+      |                               | as lpAmount / debt            | trigger liquidation
```
Comment (cosmic-vagabond) [inline x/leveragelp/spec/incentivized_liquidation_system.md:25]: in the case of a SL it triggers a position closure not a liquidation

File: x/leveragelp/spec/incentivized_liquidation_system.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+|  Earns     |                  | Manages    |                  | Earns       |                  
+|  Interest  |                  | Position   |                  | Reward      |                  
++------------+                  +------------+                  +-------------+                   
+```
+
+## Explanation
+
+- **Depositors** provide liquidity to the system and earn interest.
+- **Borrowers** take leveraged positions by borrowing from the pool.
+- **Liquidators** and **Bots** monitor positions and execute liquidations when necessary.
```
Comment (cosmic-vagabond) [inline x/leveragelp/spec/incentivized_liquidation_system.md:40]: we might wanna explain here what we mean by `Bots`

File: x/leveragelp/spec/incentivized_liquidation_system.md (hunk 1 above)
Comment (amityadav0) [reply x/leveragelp/spec/incentivized_liquidation_system.md:9]: makes, sense
watchers seems good, I will add a definition below too

File: x/leveragelp/spec/incentivized_liquidation_system.md (hunk 2 above)
Comment (amityadav0) [reply x/leveragelp/spec/incentivized_liquidation_system.md:25]: updating

File: x/leveragelp/spec/incentivized_liquidation_system.md (hunk 3 above)
Comment (amityadav0) [reply x/leveragelp/spec/incentivized_liquidation_system.md:40]: ok

**Comment by amityadav0** [comment]
Updated, please review @cosmic-vagabond

**Review by cosmic-vagabond — APPROVED** [review]

**Pull request merged**